app/controllers/api/conversion_hosts_controller.rb

```diff
@@ -1,5 +1,103 @@
 module Api
   class ConversionHostsController < BaseController
     include Subcollections::Tags
+
+    # Whitelist of valid resource types
+    VALID_TYPES = {
+      'ManageIQ::Providers::Openstack::CloudManager::Vm' => ManageIQ::Providers::Openstack::CloudManager::Vm,
+      'ManageIQ::Providers::Redhat::InfraManager::Host'  => ManageIQ::Providers::Redhat::InfraManager::Host
+    }
+
+    # Create a conversion host and enable it. This operation will run as an
+    # MiqTask.
+    #
+    # Both the 'resource_type' and 'resource_id' are mandatory arguments,
+    # and the 'resource_type' must be either 'Host' or 'VmOrTemplate'.
+    #
+    # You may optionally pass in 'param_v2v_vddk_package_url' or 'auth_user'
+    # arguments as well.
+    #
+    # POST /api/conversion_hosts {
+    #   "name": "some_name",
+    #   "resource_type": "Host",
+    #   "resource_id": "7"
+    #   "param_v2v_vddk_package_url": "some_url"
+    #   "auth_user": "some_user"
+    # }
+    #
+    def create_resource(type, id, data)
+      raise BadRequestError, "resource_id must be specified" unless data['resource_id']
+      raise BadRequestError, "resource_type must be specified" unless data['resource_type']
+      raise BadRequestError, "invalid resource_type #{data['resource_type']}" unless VALID_TYPES[data['resource_type']]
+
+      resource_type = VALID_TYPES[data['resource_type']]
+      collection_type = resource_type.table_name
+
+      # The 'auth_user' param must be deleted since the model will otherwise
+      # pass the data hash directly as params to ConversionHost.new.
+      auth_user = data.delete('auth_user')
+      resource = resource_search(data['resource_id'], resource_type.to_s, collection_class(collection_type))
+
+      data['resource'] = resource
+
+      api_action(type, id) do
+        begin
+          message = "Enabling resource id:#{resource.id} type:#{resource.type}"
+          task_id = ConversionHost.enable_queue(data, auth_user)
+          action_result(true, message, :task_id => task_id)
+        rescue => err
+          action_result(false, err.to_s)
+        end
+      end
+    end
+
+    # Disable the conversion host role by installing the conversion host module
+    # and running the conversion host playbook that disables it. This operation
+    # run as an MiqTask.
+    #
+    # You may optionally provide an 'auth_user' parameter.
+    #
+    # POST /api/conversion_hosts/:id { "action": "disable" }
+    # POST /api/conversion_hosts/:id { "action": "disable", "auth_user": "someone" }
+    #
+    # This differs from the DELETE action in that it returns a response body.
+    #
+    def disable_resource(type, id, data)
```

Comment:
I still think this is a misleading API

Reply:
I understand, but I'm just following the core methods really.

Reply:
My only concern is that core methods are not exposed to end users, per se, and APIs are.  And once the API is out there, you have to do the backwards compatibility dance to undo it.  IMO, if there is an issue with the name, we should choose a name that doesn't confuse end users, use that in the API, then possibly refactor the backend.

Reply:
If I'm reading this all correctly, does "disable" actually do a "destroy" under the covers?  If so, calling it "disable" is just wrong, as disable implies you can eventually "enable" the object with the same id later on.

> This differs from the DELETE action in that it returns a response body.
> ...
> This differs from the POST action in that it does not return a response body.

Why do we care about the distinction?  My first thought is that the POST with an "action" of "delete" (as opposed to  "disable") should be equivalent to the DELETE action and both should return the response body

Reply:
yeah, agree that enable/disable is confusing here. we can use create/destroy here and refactor the backend name.

Reply:
Alright, so really then it's just a matter of deleting the `disable` controller method altogether and letting the `delete_resource` work as for both.

Reply:
I'm not even sure we need to refactor the backend name.  enable/disable is the backend terminology and create/delete is the API, and that seems pretty straightforward.  As long as we don't add a 3rd option to the API (which is what's happening in this PR - create/delete/disable), then I think we are ok.

Reply:
Ok updated.